1_network/src/reach_classification.R

```diff
@@ -0,0 +1,136 @@
+classify_segments <- function(out_ind, network_ind, reservoir_ind) {
+  # read in data
+  res <- readRDS(sc_retrieve(reservoir_ind, 'getters.yml'))
+  network <- readRDS(sc_retrieve(network_ind, 'getters.yml'))
+
+  # group by reservoir
+  each_res <- ungroup(res) %>%
+    filter(!is.na(GRAND_ID))
+
+  # classify reaches by in reservor
+  # for each reservoir:
+  # find unique IDs
+  test <- filter(each_res, GRAND_ID %in% unique(each_res$GRAND_ID)[4])
+  touching_reservoir <- unique(test$subseg_seg)
+
+  # first find segments that are touching reservoir
+  # and classify in relation to dam/reservoir
+  segs <- each_res %>%
+    mutate(frac_overlap = as.numeric(frac_res_overlap)) %>%
+    mutate(type = case_when(
+      frac_overlap == 1 ~ 'within_reservoir',
+      is.na(inlet_intersects) & is.na(outlet_intersects) ~ 'contains_reservoir',
+      frac_overlap < 1 & !is.na(inlet_intersects) ~ 'reservoir_outlet_reach',
+      frac_overlap < 1 & !is.na(outlet_intersects) ~ 'reservoir_inlet_reach'
+    )) %>%
+    mutate(dist_down_reservoir = case_when(
+      type %in% c('within_reservoir', 'reservoir_inlet_reach') ~ units::as_units(0, 'm'),
+      type %in% c('reservoir_outlet_reach', 'contains_reservoir') ~ dist_outlet_to_dam
+    )) %>%
+    group_by(GRAND_ID) %>%
+    mutate(number_reaches = n()) %>% ungroup()
+
+  # collapse segs down to a single row per segment
+  # some segments have two reservoirs
+  # in this case, take the min distance downstream
+  segs_single <- segs %>%
+    group_by(subseg_seg, seg_id_nat, to_seg, from_segs) %>%
+    summarize(number_reservoirs = n(),
+              GRAND_ID = paste(GRAND_ID, collapse = '; '),
+              frac_overlap = sum(frac_overlap),
+              type = ifelse(grepl('contains', paste(unique(type), collapse = '')), 'contains_reservoir', unique(type)),
+              dist_down_reservoir = min(dist_down_reservoir)) %>%
+    ungroup()
+
+  res <- ungroup(res)
+
+  # grab downstream reaches
+  # keep track of which reservoirs the downstream segments are associated with
+
+  downstream <- filter(segs_single, type %in% c('contains_reservoir', 'reservoir_outlet_reach')) %>%
+    filter(!to_seg %in% unique(segs_single$subseg_seg))
+
+  # calculate combined distances
+
+  downstream1 <- downstream %>% pull(to_seg)
+
+  # downstream1 distance
```

Comment:
Are the commented blocks still needed?

Reply:
Just leaving in case we decide at some point to calculate distance to reservoir.